internal/services/compute/managed_disk_resource.go

```diff
@@ -514,6 +536,24 @@ func resourceManagedDiskUpdate(d *pluginsdk.ResourceData, meta interface{}) erro
 		}
 	}
 
+	if burstingEnabled {
+		switch storageAccountType {
+		case string(compute.StorageAccountTypesPremiumLRS):
+		case string(compute.StorageAccountTypesPremiumZRS):
+		default:
+			return fmt.Errorf("`bursting_enabled` can only be set to true when `storage_account_type` is set to `Premium_LRS` or `Premium_ZRS`")
+		}
+
+		if diskSizeGB != 0 && diskSizeGB <= 512 {
```

Comment:
the API appears to also support this for smaller Standard SSD's too:

> An on-demand bursting model, where the disk bursts whenever its needs exceed its current capacity. This model incurs additional charges anytime the disk bursts. On-demand bursting is only available for premium SSDs larger than 512 GiB.
> A credit-based model, where the disk will burst only if it has burst credits accumulated in its credit bucket. This model does not incur additional charges when the disk bursts. Credit-based bursting is only available for premium and standard SSDs 512 GiB and smaller.

https://docs.microsoft.com/en-us/azure/virtual-machines/disk-bursting#disk-level-bursting

As such I think this check could ultimately be removed?

Reply:
Same as the check in create